### ISSUE TITLE
Construction from a `Ptr{UInt8}`

### ISSUE THREAD

**Issue by oxinabox** [JuliaData/InlineStrings.jl#8]
Base provides `unsafe_string(::Ptr{UInt8}, num_bytes)::String`
It would be nice to have a similar function for creating a `InlineString` that avoids ever allocating memory.

**Comment by quinnj** [comment]
yeah, this would be trivial to add; if someone wants to take a stab at it, we already have the InlineString constructor from `bytes::Vector{UInt8}` plus `pos` and `len` that basically does this.

**Comment by oxinabox** [comment]
The hard bit though is naming

`unsafe_inlinestring`?
